src/TestFramework/TestFramework/Assertions/CollectionAssert.cs

```diff
@@ -418,9 +418,23 @@ public static void IsSubsetOf([NotNull] ICollection? subset, [NotNull] ICollecti
     {
         Assert.CheckParameterNotNull(subset, "CollectionAssert.IsSubsetOf", "subset", string.Empty);
         Assert.CheckParameterNotNull(superset, "CollectionAssert.IsSubsetOf", "superset", string.Empty);
-        if (!IsSubsetOfHelper(subset, superset))
+        Tuple<bool, ICollection<object?>> isSubsetValue = IsSubsetOfHelper(subset, superset);
+        if (!isSubsetValue.Item1)
         {
-            Assert.ThrowAssertFailed("CollectionAssert.IsSubsetOf", Assert.BuildUserMessage(message, parameters));
+            string returnedSubsetValueMessage = string.Empty;
+            foreach (object? item in isSubsetValue.Item2)
+            {
+                returnedSubsetValueMessage = returnedSubsetValueMessage + " " + Convert.ToString(item, CultureInfo.InvariantCulture) + ",";
+            }
+
+            // Remove the last trailing comma
+            if (!string.IsNullOrEmpty(returnedSubsetValueMessage))
+            {
+                returnedSubsetValueMessage = returnedSubsetValueMessage.TrimEnd(',').Trim();
+            }
+
+            returnedSubsetValueMessage = "Element(s) {" + returnedSubsetValueMessage.Trim() + "} is/are not a subset of the superset";
```

Comment:
This string needs to be localized. To do that, you need to add it to resx file. After that, when you build, the build will automatically update Xlf files and mark them as need translation. The actual translation will be brought in an automated PR later on (you don't need to care about the actual translations).